packages/browser/src/node/plugin.ts

```diff
@@ -382,11 +382,13 @@ export default (parentServer: ParentBrowserProject, base = '/'): Plugin[] => {
     {
       name: 'vitest:browser:in-source-tests',
       transform(code, id) {
+        const filename = cleanUrl(id)
         const project = parentServer.vitest.getProjectByName(parentServer.config.name)
-        if (!project._isCachedTestFile(id) || !code.includes('import.meta.vitest')) {
+
+        if (!project._isCachedTestFile(filename) || !code.includes('import.meta.vitest')) {
```

Comment:
This change is needed so that `/Users/x/vitest/test/coverage-test/fixtures/src/in-source.ts?import&browserv=1747838764408` is recognized as cached test file.